app/views/questionnaire.py

```diff
@@ -191,6 +193,20 @@ def get_block_handler(
     answer_store = questionnaire_store.answer_store
 
     current_location = Location(block_id, list_name, list_item_id)
+    try:
+        current_section_id = schema.section(block_id=block_id)['id']
+    except (AttributeError, TypeError):
+        current_section_id = None
```

Comment:
Why would this ever fail? The block_id is validated and must exist, and must exist within a section.

Reply:
I was getting TypeErrors in the following tests:

TestQuestionnaireListCollector.test_invalid_block
TestQuestionnaireEndpointRedirects.test_get_invalid_questionnaire_location_redirects_to_latest

Reply:
Those errors should be caught in the `validate_location` method rather than here.